tests/integration/workflows/helpers/calls.py

```diff
@@ -193,3 +193,28 @@ def core_record():
     assert categories
 
     return json_data, categories
+
+
+def do_validation_callback(app, data, malformed=False):
+    """Do a validation callback request.
+
+       Note:
+           If ``malformed`` is true it will make the payload
+           invalid, by removing two required ``keys``; ``id``
+           and ``_extra_data``.
+    """
+    client = app.test_client()
+    payload = {
+        "metadata": data.data,
+    }
+
+    if not malformed:
```

Comment:
Maybe do it the other way around (as you state in the comment):
```python
if malformed:
    del payload['id']
    del payload['_extra_data']
```

That also get rid of the `if not` thingie.